src/Microsoft.ApiDesignGuidelines.Analyzers/Core/IdentifiersShouldNotMatchKeywords.cs

```diff
@@ -95,28 +96,28 @@ public sealed class IdentifiersShouldNotMatchKeywordsAnalyzer : DiagnosticAnalyz
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            // TODO: Make this analyzer thread-safe.
-            //analysisContext.EnableConcurrentExecution();
-
+            analysisContext.EnableConcurrentExecution();
             analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
 
-            var keywordNamedNamespaces = new HashSet<string>();
+            analysisContext.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var keywordNamedNamespaces = new ConcurrentDictionary<string, string>();
```

Comment:
May I ask why?

`ConcurrentBag` is unordered and allows duplicates, so I assume adding is fast. Checking if an item exists would need to traverse the entire collection (possibly stepping over duplicates). On the other hand, lookups in ``ConcurrentDictionary`2`` are expected to be fast, because entries are keyed by their hashes. Adding may be slower perhaps, because of tree maintenance.

In this scenario, lookups are the common execution path, while adding entries is rare. Therefore I expect that ``ConcurrentDictionary`2`` is a better fit.

I chose ``ConcurrentDictionary`2`` based on recommendations at http://stackoverflow.com/questions/18922985/concurrent-hashsett-in-net-framework.